ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -848,7 +846,7 @@ getModIfaceFromDiskAndIndexRule recorder =
       hie_loc = Compat.ml_hie_file $ ms_location ms
   fileHash <- liftIO $ Util.getFileHash hie_loc
   mrow <- liftIO $ withHieDb (\hieDb -> HieDb.lookupHieFileFromSource hieDb (fromNormalizedFilePath f))
-  hie_loc' <- liftIO $ traverse (makeAbsolute . HieDb.hieModuleHieFile) mrow
```

Comment:
The doc said hieModuleHieFile would return a full path, I guess it is not necessary to absolute it.

Reply:
Perhaps we should also introduce an abstraction that tracks whether we expect a path to be absolute or relative?

Reply:
Seems like a good idea, some places we might need this.
path from [ghc, hiedb, lsp client]. 

The ones from ghc should depend on how we load the file into `ghc` in sessionloader?

Reply:
It is quite a mess in our codebase, we might need to open an issue and try to sweep and standardize this.